### PR TITLE
feat: integrate post-request script to the engine - INS-3785,INS-3786

### PR DESCRIPTION
**Pull request by ihexxa**
### Background
Integrating the post-request script to the engine, will start another pr for renaming related functions and variables.

### Changes:

- [x] integrate post-request script to the engine 
- [x] transform insomnia response to the script response format
- [x] tests added in: #7331 
- [x] target branch: [feature-post-request-script](https://github.com/Kong/insomnia/tree/feature-post-request-script)

### CODE REVIEW

File: packages/insomnia/src/network/network.ts [hunk 1]
```diff
@@ -74,13 +74,15 @@ export const fetchRequestData = async (requestId: string) => {
 };
 
 export const tryToExecutePreRequestScript = async (
```
Comment (jackkav) [inline packages/insomnia/src/network/network.ts]: its cleaner to use two functions than reuse in a implicit way

File: packages/insomnia/src/ui/routes/request.tsx [hunk 2]
```diff
@@ -500,6 +496,45 @@ export const sendAction: ActionFunction = async ({ request, params }) => {
...
+  }
+  if (mutatedContext.baseEnvironment) {
+    await models.environment.update(
+      baseEnvironment,
+      {
+        data: mutatedContext.baseEnvironment.data,
+        dataPropertyOrder: mutatedContext.baseEnvironment.dataPropertyOrder,
+      }
+    );
+  }
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/request.tsx:533]: ideally the postman structures should all be transformed to insomnia structures before leaving the sdk.

File: packages/insomnia-sdk/src/objects/response.ts [hunk 3]
```diff
@@ -183,3 +185,71 @@ export class Response extends Property {
...
+export async function toScriptResponse(
+    originalRequest: Request,
+    partialInsoResponse: Awaited<ReturnType<typeof sendCurlAndWriteTimeline>>,
+): Promise<Response | undefined> {
+    if (partialInsoResponse.error) {
+        // response basically doesn't contain anything
+        return undefined;
+    }
+
+    // TODO: improve the type from sendCurlAndWriteTimeline a bit
```
Comment (jackkav) [inline packages/insomnia-sdk/src/objects/response.ts]: Lets address this together, might help to simplify this code.

Comment (ihexxa) [reply packages/insomnia-sdk/src/objects/response.ts]: Let me also make a change for this.

File: packages/insomnia/src/network/network.ts (hunk 1 above)
Comment (ihexxa) [reply packages/insomnia/src/network/network.ts]: I checked the common part and most of logic seems able to be shared, for avoid duplication, I would propose keep the common part and I've added separated functions for pre and post script for improving typing.

File: packages/insomnia/src/ui/routes/request.tsx (hunk 2 above)
Comment (ihexxa) [reply packages/insomnia/src/ui/routes/request.tsx:533]: Hm Im not pretty sure and environments seem transformed.

File: packages/insomnia/src/network/network.ts [hunk 4]
```diff
@@ -101,7 +104,7 @@ export const tryToExecutePreRequestScript = async (
       }, timeout + 1000);
     });
     const preRequestPromise = cancellableRunPreRequestScript({
```
Comment (jackkav) [inline packages/insomnia/src/network/network.ts]: preRequest -> script

File: packages/insomnia-sdk/src/objects/response.ts [hunk 5]
```diff
@@ -183,3 +184,64 @@ export class Response extends Property {
...
+            bodyPath: partialResponse.bodyPath,
+            bodyCompression: partialResponse.bodyCompression,
+        });
+
+        if (readResponseResult.error) {
+            throw Error(`Failed to read body: ${readResponseResult.error}`);
+        } else {
+            responseBody = readResponseResult.body;
+        }
+    }
```
Comment (jackkav) [inline packages/insomnia-sdk/src/objects/response.ts]: consider extracting this async code from the transform, just to split the transform from the response file parsing.

File: packages/insomnia/src/network/network.ts (hunk 4 above)
Comment (ihexxa) [reply packages/insomnia/src/network/network.ts]: As renaming would introduce noice, will do this in another PR.

**Review by filfreire — APPROVED** [review]

**Pull request merged**